src/functions/notifications/scheduledDeactivateShortUrl.ts

```diff
@@ -0,0 +1,19 @@
+import { ScheduledDeactivateShortUrlBody } from '../../types/Notification';
+import { deactivateShortUrlById } from '../../libs/shortUrl';
+import { sendDeactivationMessage } from '../../libs/sendDeactivationMessage';
+
+export const scheduledDeactivateShortUrl = async ({
+  shortUrl,
+  expireDate,
+}: ScheduledDeactivateShortUrlBody) => {
+  try {
+    console.log('schedule event', shortUrl, expireDate);
+    console.log('date now', new Date());
+
+    await deactivateShortUrlById(shortUrl.id);
+    await sendDeactivationMessage(shortUrl.shortUrl, shortUrl.userEmail);
```

Comment:
You can cover these two async functions with Promise.all([...]); 

This makes you request work in parallel and saves you a few seconds of performance. 👍🏻

Reply:
Thanks for the advice!